source/lib/analyzer/EnergyCalculation.cpp

```diff
@@ -155,50 +155,58 @@ void copyPixelValuesToBuffer(unsigned blockOffsetLuma,
 {
     if (bitDepth == 8)
     {
-        src += blockOffsetLuma;
+        src += blockOffsetLumaBytes;
         for (unsigned y = 0; y < blockSize; y++, src += srcStride)
             for (unsigned x = 0; x < blockSize; x++)
                 *(buffer++) = int16_t(src[x]);
     }
     else
     {
-        auto input = (int16_t *) (src) + blockOffsetLuma;
-        for (unsigned y = 0; y < blockSize; y++, input += srcStride / 2, buffer += blockSize)
-            std::memcpy(buffer, input, blockSize * sizeof(int16_t));
+        auto input             = (uint16_t *) (src + blockOffsetLumaBytes);
+        const auto shiftTo8Bit = (bitDepth - 8);
+        for (unsigned y = 0; y < blockSize; ++y)
+        {
+            for (unsigned x = 0; x < blockSize; ++x)
+            {
+                buffer[x] = int16_t(input[x] >> shiftTo8Bit);
+            }
+            input += srcStride / 2;
+            buffer += blockSize;
+        }
```

Comment:
We now do the simple thing. We just scale the values back down to 8 bit. While for luma this is also how ffmpeg does the conversion "up", it does something a bit smarter for chroma. But here, we will just also apply this primitive scaling for now.